### PR TITLE
Disable the SpawnerPage GPU dropdown at zero

### LINKED ISSUE #301: [KFNBC]: GPU dropdown should not be visibile if no GPUs are available
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

In the spawner page I see the `Number of GPUs` dropdown, even though the only value available is 0

### Expected Behavior

Dropdown should not be shown when GPUs are not available

### Steps To Reproduce

1. Visit spawner
2. Consume all GPUs (if needed)
3. Reload spawner

### Workaround (if any)

_No response_

### OpenShift Version

_No response_

### Openshift Version

_No response_

### What browsers are you seeing the problem on?

Firefox

### Open Data Hub Version

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by andrewballantyne**
Resolves #301

## Description
Based on UX & PM input, we will disable the dropdown if there are no options with the provided help text.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: frontend/src/pages/notebookController/screens/server/SpawnerPage.tsx [hunk 1]
```diff
@@ -418,20 +388,7 @@ const SpawnerPage: React.FC = React.memo(() => {
                 </Select>
               </FormGroup>
             )}
-            {gpuOptions && (
```
Comment (andrewballantyne) [inline frontend/src/pages/notebookController/screens/server/SpawnerPage.tsx:421]: `gpuOptions` was always an array, this check did nothing -- I stripped it out.

**Comment by andrewballantyne** [comment]
Inquiring from Kyle on how we should do a loading state. TBD

**Comment by andrewballantyne** [comment]
Loading indicator is a skeleton of the size of the dropdown... logic was run by Kyle, but the effect of the change is:
- fetch on load
- after 60 seconds, the next user click will fetch an updated GPU status (for those idle cases)
- Shows Skelton over the dropdown, but if the GPUs is zero, the message stays (call is very quick so minor flickering on a disabled input happens -- hardly noticeable with a normal connection

File: frontend/src/pages/notebookController/screens/server/GPUSelectField.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    const errorCatch = (e: Error) => {
+      if (cancelled) return;
+      setFetching(false);
+      console.error(e);
+      notification.error('Failed to fetch GPU', e.message);
+    };
+
+    fetchGPU().catch(errorCatch);
+
+    const onUserClick = (): void => {
```
Comment (cfchase) [inline frontend/src/pages/notebookController/screens/server/GPUSelectField.tsx:39]: oh, this is neat!  Perhaps in a future release we can put this higher up and add it to the global context so any component can check for "last activity"

**Review by cfchase — APPROVED** [review]
/lgtm

File: frontend/src/pages/notebookController/screens/server/GPUSelectField.tsx (hunk 2 above)
Comment (andrewballantyne) [reply frontend/src/pages/notebookController/screens/server/GPUSelectField.tsx:39]: Yeah we could definitely have a global state of when the user was last doing something. This came out of a back and forth with Kyle. Seemed like a neat feature.

**Review by DaoDaoNoCode — APPROVED** [review]
lgtm

**Pull request merged**